java-checks/src/main/java/org/sonar/java/checks/NullPointerCheck.java

```diff
@@ -409,6 +414,32 @@ private void visitRelationalNotEqualTo(BinaryExpressionTree tree) {
     }
   }
 
+  class AssignmentVisitor extends BaseTreeVisitor {
+    Set<VariableSymbol> assignedSymbols = new HashSet<>();
+
+    public Set<VariableSymbol> findAssignedVariables(Tree tree) {
+      tree.accept(this);
+      return assignedSymbols;
+    }
+
+    public Set<VariableSymbol> findAssignedVariables(List<? extends Tree> trees) {
+      for (Tree tree : trees) {
+        tree.accept(this);
+      }
+      return assignedSymbols;
+    }
+
+    @Override
+    public void visitAssignmentExpression(AssignmentExpressionTree tree) {
+      // currently we only handle assignment of the form: identifier =
+      // TODO(merciesa): array[expr] = and mse.identifier = requires advanced tracking and are left outside of the scope
+      if (tree.variable().is(Tree.Kind.IDENTIFIER)) {
+        assignedSymbols.add((VariableSymbol) semanticModel.getReference((IdentifierTreeImpl) tree.variable()));
```

Comment:
What if the variable is (for some reason) not resolved ? you will add null to the set, how will your code react to this ? You should have a testcase for this.